components/bl-spoiler-effect/src/index.js

```diff
@@ -0,0 +1,298 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const EffectTypes = {
+  BLUR      : 'blur',
+  MORPHOLOGY: 'morphology',
+  MORPHOSE  : 'morphose',
+  MUSIC     : 'music',
+  RIPPLES   : 'ripples',
+  SMOKE     : 'smoke',
+  SNOW      : 'snow',
+  TURBULENCE: 'turbulence',
+};
+const EffectViews = {
+  [EffectTypes.BLUR]      : BlurEffect,
+  [EffectTypes.MORPHOLOGY]: MorphologyEffect,
+  [EffectTypes.MORPHOSE]  : MorphoseEffect,
+  [EffectTypes.MUSIC]     : MusicEffect,
+  [EffectTypes.RIPPLES]   : RipplesEffect,
+  [EffectTypes.SMOKE]     : SmokeEffect,
+  [EffectTypes.SNOW]      : SnowEffect,
+  [EffectTypes.TURBULENCE]: TurbulenceEffect,
+};
+const DefaultValues = {
+  OPACITY       : 0.5,
+  BASE_FREQUENCY: 200,
+  MIN_RADIUS    : 1,
+  MAX_RADIUS    : 7,
+  DESIRE_DENSITY: 0.0002,
+};
+
+export default function SpoilerEffectComponent({ component, elRef, eventHandlers, pods }) {
+  const { style, classList, display, effect, background, applyEffect } = component;
+  const { onClick, onMouseEnter, onMouseMove, onMouseLeave } = eventHandlers;
+
+  const [isApplied, setIsApplied] = useState(applyEffect);
+
+  const handleOnClick = useCallback(() => {
+    setIsApplied(!isApplied);
```

Comment:
```suggestion
    setIsApplied(v => !v);
```

and remove `isAplied` from deps